multiversx_sdk_core/transaction_factories/smart_contract_factory_test.py

```diff
@@ -14,83 +14,63 @@ class TestSmartContract:
 
     def test_smart_contract_deploy(self):
         sender = Address.from_bech32("erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th")
-        guardian = Address.from_bech32("erd1spyavw0956vq68xj8y4tenjpq2wd5a9p2c6j8gsz7ztyrnpxrruqzu66jx")
-        nonce = 777
         contract = Path(__file__).parent.parent / "testutils" / "testdata" / "adder.wasm"
         gas_limit = 6000000
         args = [0]
 
-        transaction = self.factory.deploy(
-            deployer=sender,
-            nonce=nonce,
-            bytecode_path=contract,
+        transaction = self.factory.create_transaction_intent_for_deploy(
```

Comment:
Perhaps we should also rename the test functions, e.g. `test_create_transaction_intent_for_deploy` (optionally / debatable).